Rkhan/tests and refactoring

RizaHKhan opened the pull request:
#4 
#1 

Sorry this is such a large PR :(. Kill two birds with one stone.

RizaHKhan committed:
added test for button

RizaHKhan committed:
adding test for base input component

RizaHKhan committed:
base text component

RizaHKhan committed:
moving components around

RizaHKhan committed:
accomodations for new component

JadRizk commented:
Hey man ! Sure don't worry about the PR size 😅

Okay so the idea of having a `baseInput` component is have a common input across the whole application(input field, checkbox, ...)

I would suggest keeping the BaseInput as it was - simple input (atom) - and build a new component called a `textField` component that could handle a the Input, Label and, error tag.

RizaHKhan committed:
additional tests for button

RizaHKhan committed:
added settings route

RizaHKhan committed:
added route to setting page

RizaHKhan committed:
added ability to disable button based on prop

RizaHKhan committed:
added to make to see all of our changes

RizaHKhan committed:
re-added base input as it was and renamed new component as textfield

RizaHKhan committed:
renamed component as styleguide

RizaHKhan committed:
test for textfield

RizaHKhan commented:
@JadRizk Agreed!

BaseInput reverted to original version and added TextField component (also an atom, though it has a little bit more functionality).

Also, you will notice I added a `StyleGuide` file where we can showcase all of our various components and how they look. I picked this up from my current place of work and I find it quite useful. I actually found an issue that I've added as an issue.

Also added more tests :)

Let me know if its all good, thanks!

RizaHKhan committed:
added another expect to button test

RizaHKhan committed:
removing unused item

RizaHKhan committed:
deleting setting file

JadRizk commented on tests/unit/TextField.spec.ts:
```diff
@@ -0,0 +1,28 @@
+import TextField from '@/components/atoms/TextField.atom.vue'
+
+describe('TextField', () => {
+  const id = 'greatestId'
+  const name = 'GhenghisKhan'
```
Consider adding the following fields to remove existing warnings

```suggestion
  const id = 'greatestId'
  const name = 'GhenghisKhan'
  const label = 'TextField-Label'
  const iconName = 'IconName'
  const modelValue = 'modelValue'
```

RizaHKhan replied:
props added

JadRizk commented on tests/unit/TextField.spec.ts:
```diff
@@ -0,0 +1,28 @@
+  const name = 'GhenghisKhan'
+
+  it('should emit a value', async () => {
+    const wrapper = shallowMount(TextField, {
+      props: { id, name },
+    })
```
```suggestion
   const wrapper = shallowMount(TextField, {
      props: { id, name, label, iconName, modelValue },
    })
```

JadRizk commented on tests/unit/TextField.spec.ts:
```diff
@@ -0,0 +1,28 @@
+
+  it('emits on-blur when on blur', async () => {
+    const wrapper = shallowMount(TextField, {
+      props: { id, name },
```
```suggestion
      props: { id, name, label, iconName, modelValue },
```

RizaHKhan committed:
Merge branch 'master' into rkhan/tests_and_refactoring

RizaHKhan committed:
readded original component

RizaHKhan committed:
deleted new component and adjusted various items to accomodate

RizaHKhan commented on src/components/atoms/TextField.atom.vue:
```diff
@@ -1,33 +1,34 @@
-      :errorMode="validationStatus.$error"
-      @on-blur="handleOnBlur"
-      @update:modelValue="emit('update:modelValue', $event)"
+      @input="emit('update:modelValue', $event.target.value)"
```
reverted to original version. When I was creating it, I felt that the original version seemed better anyways.

RizaHKhan committed:
some import and naming adjustments

RizaHKhan committed:
added additional props for test

RizaHKhan committed:
additional props for both tests

RizaHKhan commented:
There is an issue in th test:

![image](https://user-images.githubusercontent.com/51720003/114283832-deef0b80-9a19-11eb-9500-7d36a0b8b5f3.png)
Since we are using aliases for our imports, its throwing the jest tests off. I've searched high and low for a resolution but nothing. Perhaps you would know?
